src/catnap/mod.rs

```diff
@@ -413,7 +414,23 @@ impl CatnapLibOS {
         let boxed_future: Box<dyn Any> = self.runtime.take(handle).as_any();
         let boxed_concrete_type: Operation =
             *boxed_future.downcast::<Operation>().expect("Wrong type!");
-        boxed_concrete_type.get_result()
+
+        let (qd, new_qd, new_fd, qr): (QDesc, Option<QDesc>, Option<RawFd>, OperationResult) =
+            boxed_concrete_type.get_result();
+
+        // Handle accept operation.
+        if let Some(new_qd) = new_qd {
+            // Associate raw file descriptor with queue descriptor.
+            if let Some(new_fd) = new_fd {
+                assert_eq!(self.sockets.insert(new_qd, new_fd).is_none(), true);
+            }
+            // Release entry in queue table.
```

Comment:
This is a cleanup after some failure?

Reply:
Yes it is. I'm gonna move the comment to the line bellow.